app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginFragment.java

```diff
@@ -23,11 +23,11 @@
 
 public class LoginFragment extends Fragment implements LoginView {
     FragmentLoginBinding binding;
-    private SharedPreferences preferences;
-    private SharedPreferences.Editor editor;
     private ProgressDialog dialog;
     int showPassword;
     LoginPresenter loginPresenter;
+    SharedPreference sharedPreference;
+    ISharedPreferenceInterface iSharedPreferenceInterface;
```

Comment:
Pass the implementation to LoginPresenter it will be something like that 
LoginPresenter(this, new SharedPreferenceImpl(context))

And for the context, it's better to user `ApplicationContext` rather than `ActivityContext` to avoid leaking.